scipy/stats/_continuous_distns.py

```diff
@@ -517,15 +517,19 @@ def _entropy(self):
 
 class FitDataError(ValueError):
     # This exception is raised by, for example, beta_gen.fit when both floc
-    # and fscale  are fixed and there are values in the data not in the open
-    # interval (floc, floc+fscale).
-    def __init__(self, distr, lower, upper):
-        self.args = (
-            "Invalid values in `data`.  Maximum likelihood "
-            "estimation with {distr!r} requires that {lower!r} < x "
-            "< {upper!r} for each x in `data`.".format(
-                distr=distr, lower=lower, upper=upper),
-        )
+    # and fscale are fixed and there are values in the data not in the open
+    # interval (floc, floc+fscale), or with a custom message.
+    def __init__(self, distr, lower, upper, **kwds):
+        msg = kwds.pop('msg', None)
+        if msg:
+            self.args=(f"{msg}",)
+        else: 
+            self.args = (
+                "Invalid values in `data`.  Maximum likelihood "
+                "estimation with {distr!r} requires that {lower!r} < x "
```

Comment:
@WarrenWeckesser this isn't specific to this PR - shouldn't this always be (x-loc)/scale? Is the user expected to  understand that?